examples/chip-tool/commands/clusters/ClusterCommand.h

```diff
@@ -117,7 +127,28 @@ class ClusterCommand : public InteractionModelCommands, public ModelCommand, pub
                 mError = error;
                 return;
             }
+            if ((path.mEndpointId == chip::kRootEndpointId) && (path.mClusterId == chip::app::Clusters::IcdManagement::Id) &&
+                (path.mCommandId == chip::app::Clusters::IcdManagement::Commands::RegisterClient::Id))
+            {
+                chip::TLV::TLVReader counterTlvReader;
+                counterTlvReader.Init(*data);
+                chip::app::Clusters::IcdManagement::Commands::RegisterClientResponse::DecodableType value;
+                CHIP_ERROR err = chip::app::DataModel::Decode(counterTlvReader, value);
+                if (CHIP_NO_ERROR != err)
+                {
+                    ChipLogError(chipTool, "Failed to decode ICD counter: %" CHIP_ERROR_FORMAT, err.Format());
+                    return;
+                }
+
+                chip::app::ICDClientInfo clientInfo;
+                clientInfo.peer_node         = mScopedNodeId;
+                clientInfo.monitored_subject = mMonitoredSubject;
+                clientInfo.start_icd_counter = value.ICDCounter;
+
+                ModelCommand::AddICDEntry(clientInfo, chip::ByteSpan(mICDSymmetricKey));
```

Comment:
Why does this need the `ModelCommand` prefix?

Reply:
removed, thanks